lib/app.js

```diff
@@ -102,15 +101,35 @@ class App {
       if      (shell.test('-e', oldTrufflePath)) truffleConfig = reqCwd.silent(oldTrufflePath);
       else if (shell.test('-e', newTrufflePath)) truffleConfig = reqCwd.silent(newTrufflePath);
 
+      const coverageNetwork = {
+        host: 'localhost',
+        network_id: '*',
+        port: this.port,
+        gas: gasLimitHex,
+        gasPrice: gasPriceHex
+      };
+
       // Coverage network opts specified: use port if declared
       if (truffleConfig && truffleConfig.networks && truffleConfig.networks.coverage) {
         this.port = truffleConfig.networks.coverage.port || this.port;
         this.network = '--network coverage';
```

Comment:
The `this.network` assignment should come out of the `if` block here so it's always set. Maybe just right before this piece of logic. It gets appended to the `truffle test` command later and I think the unit-tests are failing because they're not running `truffle test --network coverage` as they should.